mmocr/visualization/kie_visualizer.py

```diff
@@ -0,0 +1,505 @@
+        classes_image = self._draw_labels(self, classes_image, bbox_classes,
+                                          bboxes)
+        if polygons:
+            classes_image = self._draw_polygons(
+                self, classes_image, polygons, colors=PALETTE)
+        else:
+            classes_image = self._draw_bboxes(
+                self, classes_image, bboxes, colors=PALETTE)
+
+        edge_image = None
+        if is_openset:
+            edge_image = np.full(empty_shape, 255, dtype=np.uint8)
+            edge_image = self._draw_edge_label(edge_image, edge_labels, bboxes,
+                                               texts, arrow_colors)
+        cat_image = []
+        for i in [image, text_image, classes_image, edge_image]:
+            if i is not None:
+                cat_image.append(i)
+        return np.concatenate(cat_image, axis=1)
+
+    def add_datasample(self,
+                       name: str,
+                       image: np.ndarray,
+                       data_sample: Optional['KIEDataSample'] = None,
+                       draw_gt: bool = True,
+                       draw_pred: bool = True,
+                       show: bool = False,
+                       wait_time: int = 0,
+                       pred_score_thr: float = None,
+                       out_file: Optional[str] = None,
+                       step: int = 0) -> None:
+        """Draw datasample and save to all backends.
+
+        - If GT and prediction are plotted at the same time, they are
+        displayed in a stitched image where the left image is the
+        ground truth and the right image is the prediction.
+        - If ``show`` is True, all storage backends are ignored, and
+        the images will be displayed in a local window.
+        - If ``out_file`` is specified, the drawn image will be
+        saved to ``out_file``. This is usually used when the display
+        is not available.
+
+        Args:
+            name (str): The image identifier.
+            image (np.ndarray): The image to draw.
+            data_sample (:obj:`TextDetDataSample`, optional):
+                TextDetDataSample which contains gt and prediction. Defaults
+                    to None.
+            draw_gt (bool): Whether to draw GT TextDetDataSample.
+                Defaults to True.
+            draw_pred (bool): Whether to draw Predicted TextDetDataSample.
+                Defaults to True.
+            show (bool): Whether to display the drawn image. Default to False.
+            wait_time (float): The interval of show (s). Defaults to 0.
+            out_file (str): Path to output file. Defaults to None.
+            pred_score_thr (float): The threshold to visualize the bboxes
+                and masks. Defaults to 0.3.
+            step (int): Global step value to record. Defaults to 0.
+        """
+        gt_img_data = None
+        pred_img_data = None
+
+        if draw_gt:
+            gt_bboxes = data_sample.gt_instances.bboxes
```

Comment:
```suggestion
            gt_bboxes = data_sample.gt_instances.get('bboxes', None)
```
Do the change you have taken my previous suggestions in `_draw_instances`, same for line 353